libcudacxx/include/cuda/std/__cccl/compiler.h

```diff
@@ -11,25 +11,33 @@
 #ifndef __CCCL_COMPILER_H
 #define __CCCL_COMPILER_H
 
-// Determine the host compiler
+// Determine the host compiler and its version
 #if defined(__INTEL_LLVM_COMPILER)
 #  define _CCCL_COMPILER_ICC_LLVM
+#  define _CCCL_COMPILER_ICC_LLVM_VERSION __INTEL_LLVM_COMPILER
```

Comment:
Remark: The Intel LLVM compiler is not a supported compiler of CTK (see: https://docs.nvidia.com/cuda/cuda-installation-guide-linux/index.html#id11). Support can we removed as well. Again: consider a separate PR.

Reply:
I removed all references to Intel LLVM compiler. I didn't create a new PR because it appears just a couple of times. Anyway, thanks for the suggestion. I try to split my changes in smaller PR in the future